pkg/kubelet/kubeletconfig/util/files/files.go

```diff
@@ -63,34 +63,54 @@ func EnsureFile(fs utilfs.Filesystem, path string) error {
 	return file.Close()
 }
 
-// ReplaceFile replaces the contents of the file at `path` with `data` by writing to a tmp file in the same
-// dir as `path` and renaming the tmp file over `path`. The file does not have to exist to use ReplaceFile.
-func ReplaceFile(fs utilfs.Filesystem, path string, data []byte) error {
+// WriteTmpFile creates a temporary file at `path`, writes `data` into it, and fsyncs the file
+func WriteTmpFile(fs utilfs.Filesystem, path string, data []byte) (tmpPath string, err error) {
 	dir := filepath.Dir(path)
 	prefix := filepath.Base(path)
 
 	// create the tmp file
 	tmpFile, err := fs.TempFile(dir, prefix)
```

Comment:
I wonder if this line redefines named return values? It is probably ok, but confusing.

Reply:
It doesn't, the values make it through to the return.
I made sure to test this in the playground 😄 .
But I'll make it more explicit anyhow.